src/handlers.rs

```diff
@@ -111,21 +111,41 @@ pub fn handle_request(log: &Logger, request: &protocol::Request) -> Result<Vec<u
             };
             serialize_initgroups(groups)
         }
+
+        // There's no cache to invalidate
+        RequestType::INVALIDATE => {
+            debug!(log, "received invalidate request, ignoring");
+            Ok(vec![])
+        }
+
+        // We don't want clients to be able to shut down nsncd.
+        RequestType::SHUTDOWN => {
+            debug!(log, "received shutdown request, ignoring");
+            Ok(vec![])
+        }
+
+        // These will normally send an FD pointing to the internal cache structure,
+        // which clients use to look into the cache contents on their own.
+        // We don't cache, and we don't want clients to poke around in cache structures either.
+        // Luckily clients fall back to explicit queries if no FDs are sent over.
+        RequestType::GETFDPW
+        | RequestType::GETFDGR
+        | RequestType::GETFDHST
+        | RequestType::GETFDSERV
+        | RequestType::GETFDNETGR => {
+            debug!(log, "received GETFD* request, ignoring");
+            Ok(vec![])
+        }
+
+        // Not implemented (yet)
         RequestType::GETHOSTBYADDR
         | RequestType::GETHOSTBYADDRv6
         | RequestType::GETHOSTBYNAME
         | RequestType::GETHOSTBYNAMEv6
-        | RequestType::SHUTDOWN
         | RequestType::GETSTAT
```

Comment:
This returns statistics about the daemon - you can also mark this as will-not-implement.

Reply:
Why? I've been thinking about exposing some prometheus-style metrics on the number of requests, success rate and timing Information, by request type. That'd of course primarily be exposed via http, but if that getstat call is actually usable to expose some of these numbers, and there's some userspace tool to query and show, why not?